R/agg_scale.R

```diff
@@ -372,11 +391,13 @@ scale <- function(dt,
     cols <- paste0(col_stem, "_", c("start", "end"))
 
     # create one column to describe each interval
-    gen_name(dt, col_stem = col_stem, format = "interval")
-    data.table::setnames(dt, paste0(col_stem, "_name"), col_stem)
+    dt_intervals <- unique(dt[, .SD, .SDcols = cols])
+    gen_name(dt_intervals, col_stem = col_stem, format = "interval")
+    data.table::setnames(dt_intervals, paste0(col_stem, "_name"), col_stem)
+    data.table::setkeyv(dt_intervals, cols)
+    dt <- dt[dt_intervals, on = cols, nomatch = 0]
```

Comment:
I still think it would be helpful to have a short vignette or Hub page with advanced data.table tips.

Reply:
Started a small one on our code hub page.